tools/config-generator/udpaterelease.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	maxReleaseBranches = 4
+	// Assuming all repo with 0.14 branch also has 0.15 branch, this will be
+	// derived from the branch in following PR
+	latest = "0.15"
+)
+
+func upgradeReleaseBranchesTemplate(name string) error {
+	config := yaml.MapSlice{}
+	info, err := os.Lstat(name)
+	if err != nil {
+		return fmt.Errorf("failed stats file %q: %w", name, err)
+	}
+	content, err := ioutil.ReadFile(name)
+	if err != nil {
+		return fmt.Errorf("Cannot read file %q: %w", name, err)
+	}
+	if err = yaml.Unmarshal(content, &config); err != nil {
+		return fmt.Errorf("Cannot parse config %q: %w", name, err)
+	}
+	for i, repos := range config {
+		if repos.Key == "presubmits" {
+			continue
+		}
+
+		config[i].Value = getReposMap(repos.Value, latest)
+	}
+
+	updated, err := yaml.Marshal(&config)
+	// This shouldn't happen, just catch it in case
+	if err != nil {
+		return fmt.Errorf("failed marshal modified content: %w", err)
+	}
+	return ioutil.WriteFile(name, updated, info.Mode())
```

Comment:
This might be correct in the short-term and the WriteFile contract might not be subject to change, but we're effectively binding our contract to it.
Consider decoupling the invocation to the return value.

Reply:
Sorry but not sure what the recommendation is, did you mean make this function return string instead of make this function in place update the file?

Reply:
The recommendation is to decouple the return value from WriteFile from our function return value. Right now, both contracts are bound together and that might cause issues on the future.
I'm suggesting to reconsider that and make sure we want to *always* return whatever `WriteFile` function returns or invoke it independently and `return nil` afterwards.

Reply:
I think that's expected to let it return from WriteFile, and I think it's pretty common pattern in go